### PR TITLE
Improve Sphinx documentation

### LINKED ISSUE #41: Auto generate/build documentation
Automatically generate documentation output
- autogenerate 
- auto deploy
To be decided: location (e.g. GitHub)
To be decided: at what stage do we autogenerate/update the docs (e.g. pull requests, merge, release...)

### PR DESCRIPTION
**Pull request by pineapple-cat**
Moved documentation location to a more intuitive place:
1. Created an empty `docs/` folder at the same level as `src/` and then moved to that folder and ran `sphinx-quickbuild`.
2. Adjusted `conf.py` to include the `sphinx.ext.autodoc` extension and ran `sphinx-apidoc -o . ../src/hivpy` to set output and module paths.
3. Ran `make html` to get html docs in `_build/html`.

Still looking into how to change documentation formatting.

### CODE REVIEW

**Comment by pineapple-cat** [comment]
Made some manual changes to Sphinx's rst files to make auto-generated documentation more readable.

Further improvements that could be made:

- Changing HTML theming for even better readability.
- Changing docstring style (e.g. using [the Sphinx docstring format](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html)). May also be worth using more code blocks to improve clarity.
- Adding further documentation to undocumented functions.

File: docs/index.rst [hunk 1]
```diff
@@ -1,18 +1,15 @@
 .. HIVpy documentation master file, created by
-   sphinx-quickstart on Wed Oct  5 23:51:08 2022.
+   sphinx-quickstart on Fri Oct 14 15:57:48 2022.
    You can adapt this file completely to your liking, but it should at least
    contain the root `toctree` directive.
 
 Welcome to HIVpy's documentation!
 =================================
 
-This code simulates a model of the spread and effects of HIV in Sub-Saharan Africa.
```
Comment (pineapple-cat) [inline docs/index.rst]: Oops, this line should be kept in...

File: docs/modules.rst [hunk 2]
```diff
@@ -1,3 +1,5 @@
+:orphan:
```
Comment (pineapple-cat) [inline docs/modules.rst:1]: Prevents a warning about the document not being included in any toctree.

**Review by mmcleod89 — APPROVED** [review]
This seems to be working fine, I would like a line or two introducing the documentation to go back into index.rst. It might also be worth putting in the README a note about a) how to build the docs and b) which of the html (or other) files if the top level of the documentation that they should be clicking on, so that it's a bit easier for people to navigate.

**Pull request merged**